cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1160,31 +1160,29 @@ SpecificAndLatestVersionKeys LocalVersionedEngine::get_stream_index_map(
     return std::make_pair(specific_versions, latest_versions);
 }
 
+folly::Future<std::pair<std::optional<VariantKey>, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_metadata_future_impl(std::optional<AtomKey> key){
```

Comment:
This type is quite long maybe we can use a typedef for this. Maybe `KeyValueFuture` or something similar.

Reply:
I'm conflicted about this, as I think sometimes an alias in a signature makes it harder to read, since you have to go and look up what the type is somewhere else.